### DISCUSSION TITLE
Is it possible to return data via Ajax?

### DISCUSSION THREAD

**Discussion by anthonycalvano** [jrief/django-formset#62] [Q&A]
I've created a search form where you the user can enter search criteria and when they hit submit, the search results would, ideally, appear below the form.  I'm using FormCollectionView to create the form as the user can add multiple criteria forms before submitting.  I'm using the form_collection_valid method to get the cleaned_data so I can perform the search on the database.  But at this point I'm not sure how to return the results to the browser.  This method can only return a success_url.  Is there a way to return the results via Ajax, so the page doesn't reload?  I can save the values to the database and my success_url have parameters to reference the saved data, but I would prefer to do this all in the same view, if possible, and not have to save the data if the user doesn't need it for later.

**Comment by jrief** [comment]
This is a good point and a use-case I haven't thought of so far.

Indeed, it would be quite easy to return additional data, rather than the `success_url` inside the response JSON.
However, I currently have to button action implemented for this. It probably would be a good idea to use the `emit()` action and pass the server response to it.

If you think this helps, I might add this as a feature in version 1.1. Please create an issue for this feature request.

**Reply by anthonycalvano** [reply]
I could possibly use the emit() action.  Not entirely sure how to use it.  Would I just have to write a javascript function that takes care of the ajax request and trigger it when the button is clicked?  I can try it out.
I see that you took this and created an issue.  I'll follow that to see the outcome.  Thank you for the support!

**Reply by jrief** [reply]
> Would I just have to write a javascript function that takes care of the ajax request and trigger it when the button is clicked?

Yes. You would have to add an `eventListener()` to your own Javascript and listen for the named event.
